src/overlays/actors/ovl_En_Nb/z_en_nb.c

```diff
@@ -21,255 +169,1403 @@ const ActorInit En_Nb_InitVars = {
+        thisx->posRot.pos.z = csCmdNPCAction->startPos.z;
+        newRotY = csCmdNPCAction->rot.y;
+        thisx->shape.rot.y = newRotY;
+        thisx->posRot.rot.y = newRotY;
+    }
+}
+
+s32 func_80AB1390(EnNb* this, GlobalContext* globalCtx, u16 arg2, s32 npcActionIdx) {
+    CsCmdActorAction* csCmdNPCAction;
+    if ((globalCtx->csCtx.state != 0) &&
+        (csCmdNPCAction = globalCtx->csCtx.npcActions[npcActionIdx], csCmdNPCAction != NULL) &&
+        (csCmdNPCAction->action == arg2)) {
+        return 1;
+    }
+    return 0;
+}
+
+s32 func_80AB13D8(EnNb* this, GlobalContext* globalCtx, u16 arg2, s32 npcActionIdx) {
+    CsCmdActorAction* csCmdNPCAction;
+    if ((globalCtx->csCtx.state != 0) &&
+        (csCmdNPCAction = globalCtx->csCtx.npcActions[npcActionIdx], csCmdNPCAction != NULL) &&
+        (csCmdNPCAction->action != arg2)) {
+        return 1;
+    }
+    return 0;
+}
+
+void EnNb_SetInitialCsPosRot(EnNb* this, GlobalContext* globalCtx, s32 npcActionIdx) {
+    CsCmdActorAction* csCmdNPCAction = EnNb_GetNpcCsAction(globalCtx, npcActionIdx);
+    s16 newRotY;
+    Actor* thisx = &this->actor;
+
+    if (csCmdNPCAction != NULL) {
+        thisx->posRot.pos.x = csCmdNPCAction->startPos.x;
+        thisx->posRot.pos.y = csCmdNPCAction->startPos.y;
+        thisx->posRot.pos.z = csCmdNPCAction->startPos.z;
+        newRotY = csCmdNPCAction->rot.y;
+        thisx->shape.rot.y = newRotY;
+        thisx->posRot.rot.y = newRotY;
+    }
+}
+
+void EnNb_SetCurrentAnim(EnNb* this, AnimationHeader* animation, u8 mode, f32 transitionRate, s32 arg4) {
+    f32 frameCount = SkelAnime_GetFrameCount(&animation->genericHeader);
+    f32 playbackSpeed;
+    f32 unk0;
+    f32 fc;
+
+    if (arg4 == 0) {
+        unk0 = 0.0f;
+        fc = frameCount;
+        playbackSpeed = 1.0f;
+    } else {
+        unk0 = frameCount;
+        fc = 0.0f;
+        playbackSpeed = -1.0f;
+    }
+
+    SkelAnime_ChangeAnim(&this->skelAnime, animation, playbackSpeed, unk0, fc, mode, transitionRate);
+}
+
+void EnNb_SetCurrentAnim(EnNb* this, AnimationHeader* animation, u8 mode, f32 transitionRate, s32 arg4);
+
+void EnNb_SetChamberAnim(EnNb* this, GlobalContext* globalCtx) {
+    EnNb_SetCurrentAnim(this, &D_06009694, 0, 0, 0);
+    this->actor.shape.unk_08 = -10000.0f;
+}
+
+void EnNb_SpawnBlueWarp(EnNb* this, GlobalContext* globalCtx) {
+    Actor* thisx = &this->actor;
+    f32 posX = thisx->posRot.pos.x;
+    f32 posY = thisx->posRot.pos.y;
+    f32 posZ = thisx->posRot.pos.z;
+
+    Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_DOOR_WARP1, posX, posY, posZ, 0, 0, 0, 2);
+}
+
+void EnNb_GiveMedallion(EnNb* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    f32 posX = player->actor.posRot.pos.x;
+    f32 posY = player->actor.posRot.pos.y + 50.0f;
+    f32 posZ = player->actor.posRot.pos.z;
+
+    Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_DEMO_EFFECT, posX, posY, posZ, 0, 0, 0, 12);
```

Comment:
I think we use 0xC (hex) for actor variables (would be consistent with eg `func_809853B4` too)